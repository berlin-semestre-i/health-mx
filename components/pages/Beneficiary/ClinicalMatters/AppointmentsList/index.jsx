import React , {PureComponent} from 'react'
import {Button, List, Grid } from 'semantic-ui-react'
import styled from 'styled-components'
import ConsultationModal from '../../../../shared/ConsultationModal'

const somatometry = {
  nurse: 'Rodrigo Fernández',
  height: 169,
  pulse: 68,
  imc: 22.06,
  weight: 63,
  temperature: 36,
  status: 'En tratamiento',
  doctor: 'Mónica Dessiré López Trejo',
}

class AppointmentsList extends PureComponent {

  state = {
    open: false,
  }

  open = () => this.setState({open: true})
  close = () => this.setState({open: false})

  render() {
    const { appointment } = this.props
    const { open } = this.state

    return (
      <React.Fragment>
        <List>
          <List.Content floated="right">
            <Button secondary size="tiny" onClick={this.open}>Ver detalles</Button>
          </List.Content>
          <AppointmentData>
            <Grid>
              <Grid.Column mobile={16} tablet={5} computer={5}>
                <List.Content>
                  {appointment.date}
                </List.Content>
              </Grid.Column>
              <Grid.Column mobile={16} tablet={11} computer={11}>
                <List.Content>
                  {appointment.specialty}
                </List.Content>
              </Grid.Column>
            </Grid>
          </AppointmentData>
        </List>
        <ConsultationModal
          open={open}
          consultation={appointment}
          somatometry={somatometry}
          close={this.close}
        />
      </React.Fragment>
    )
  }
}

export default AppointmentsList

const AppointmentData = styled.div`
    & {
        padding-top: 8px;
    }
`