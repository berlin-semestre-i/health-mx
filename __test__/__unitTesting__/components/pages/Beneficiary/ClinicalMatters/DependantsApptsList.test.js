import React from 'react'
import { shallow, mount } from 'enzyme'
// eslint-disable-next-line max-len
import AppointmentsList from '../../../../../../components/pages/Beneficiary/ClinicalMatters/AppointmentsList/index'

const testAppt = {
  nss: '1',
  date: '10 Oct, 2018',
  specialty: 'Medicina familiar',
  clinic: 'General 2',
}

describe('Dependants past appointments', () => {
  it('renders', () => {
    const component = shallow(<AppointmentsList appointment={testAppt} />)
    expect(component).toMatchSnapshot()
  })
})