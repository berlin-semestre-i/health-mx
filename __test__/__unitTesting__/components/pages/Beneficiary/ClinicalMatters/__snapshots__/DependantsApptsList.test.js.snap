// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependants past appointments renders 1`] = `
<List>
  <ListContent
    floated="right"
  >
    <Link
      href="#"
    >
      <Button
        as="button"
        role="button"
        secondary={true}
        size="tiny"
      >
        Ver detalles
      </Button>
    </Link>
  </ListContent>
  <AppointmentsList__AppointmentData>
    <Grid>
      <GridColumn
        computer={5}
        mobile={16}
        tablet={5}
      >
        <ListContent>
          10 Oct, 2018
        </ListContent>
      </GridColumn>
      <GridColumn
        computer={11}
        mobile={16}
        tablet={11}
      >
        <ListContent>
          Medicina familiar
        </ListContent>
      </GridColumn>
    </Grid>
  </AppointmentsList__AppointmentData>
</List>
`;
